kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -143,28 +133,13 @@ public State next() {
     }
   }
 
-  private void updateRotationTimers() {
-    long lastRotate = time.milliseconds();
-    if (log.isDebugEnabled() && rotateIntervalMs > 0) {
-      log.debug("Update last rotation timer. Next rotation for {} will be in {}ms", tp, rotateIntervalMs);
-    }
-    if (rotateScheduleIntervalMs > 0) {
-      nextScheduledRotate = DateTimeUtils.getNextTimeAdjustedByDay(lastRotate, rotateScheduleIntervalMs, timezone);
```

Comment:
Using `rotate.interval.ms` in addition to `partition.duration.ms` will require some code in the partitioner to compare timestamps between records. And to do that we'll have to adjust time as above. Currently we do that in [getPartition](https://github.com/kkonstantine/kafka-connect-storage-common/blob/170f848b7f244a3ffda8f509342a8caf4d94b1bb/partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java#L73-L75)